spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/FeignContextBeanPostProcessor.java

```diff
@@ -43,7 +61,14 @@ public Object postProcessBeforeInitialization(Object bean, String beanName) thro
 	@Override
 	public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
 		if (bean instanceof FeignContext && !(bean instanceof TraceFeignContext)) {
-			return new TraceFeignContext(traceFeignObjectWrapper(), (FeignContext) bean);
+			FeignContext feignContext = (FeignContext) bean;
+			TraceFeignContext traceFeignContext = new TraceFeignContext(traceFeignObjectWrapper(), feignContext);
+			try {
+				traceFeignContext.setApplicationContext((ApplicationContext) PARENT.get(bean));
```

Comment:
Why not change the `FeignContext` to have a getter for the application context and then we would get rid of the reflection approach?

Reply:
That's better, will be changed